src/components/refactor/dialogs/two-windings-transformer-creation/tap-changer-pane/tap-changer-pane-taps.js

```diff
@@ -0,0 +1,368 @@
+                currentTapRows[index][createTapRuleColumn] = current;
+
+                current += interval;
+            });
+            replace(currentTapRows);
+        }
+    };
+
+    function handleImportTapRuleButton() {
+        trigger(`${tapChanger}.${LOW_TAP_POSITION}`).then((result) => {
+            if (result) {
+                setOpenImportRuleDialog(true);
+            }
+        });
+    }
+
+    const handleImportTapRule = (selectedFile, setFileParseError) => {
+        Papa.parse(selectedFile, {
+            header: true,
+            skipEmptyLines: true,
+            complete: function (results) {
+                if (results.data.length > MAX_TAP_NUMBER) {
+                    setFileParseError(
+                        intl.formatMessage(
+                            { id: 'TapPositionValueError' },
+                            { value: MAX_TAP_NUMBER }
+                        )
+                    );
+                    return;
+                }
+                let rows = results.data.map((val) => ({
+                    ...handleImportRow(val),
+                    [SELECTED]: false,
+                }));
+                if (rows && rows.length > 0) {
+                    replace(rows);
+                }
+            },
+        });
+    };
+
+    const lowTapPositionField = (
+        <IntegerInput
+            name={`${tapChanger}.${LOW_TAP_POSITION}`}
+            label="LowTapPosition"
+            formProps={{
+                disabled: disabled,
+            }}
+        />
+    );
+
+    const highTapPositionField = (
+        <IntegerInput
+            name={`${tapChanger}.${HIGH_TAP_POSITION}`}
+            label="HighTapPosition"
+            formProps={{
+                disabled: true,
+            }}
+        />
+    );
+
+    const tapPositionField = (
+        <IntegerInput
+            name={`${tapChanger}.${TAP_POSITION}`}
+            label="TapPosition"
+            formProps={{
+                disabled: disabled,
+            }}
+        />
+    );
+
+    const createRuleButton = (
+        <Tooltip
+            title={intl.formatMessage({
+                id: createRuleMessageId,
+            })}
+            placement="left"
+        >
+            <span>
+                <IconButton
+                    onClick={() => setOpenCreateRuleDialog(true)}
+                    disabled={disabled || tapFields.length === 0}
+                    sx={{
+                        position: 'absolute',
+                        right: 0,
+                        top: 0,
+                        padding: 1.5,
+                    }}
+                >
+                    <AddchartIcon />
+                </IconButton>
+            </span>
+        </Tooltip>
+    );
+
+    const completedColumnsDefinition = columnsDefinition.map((column, index) =>
+        index === columnsDefinition.length - 1
+            ? { ...column, extra: createRuleButton }
+            : column
+    );
+
```

Comment:
looping through the whole array only to change the last one sounds complex

```suggestion
    const completedColumnsDefinition = columnsDefinition;
    completedColumnsDefinition[completedColumnsDefinition.length-1] = {...completedColumnsDefinition[completedColumnsDefinition.length-1], extra: createRuleButton};

```

Reply:
You're right I change it